script.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    if block_given?
+      counter = 0
+      while counter < size
+        yield(self[counter])
+        counter += 1
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    if block_given?
+      counter = 0
+      while counter < size
+        yield(self[counter], counter)
+        counter += 1
+      end
+    end
+    self
+  end
+
+  def my_select
+    if block_given?
+      arr = []
+      my_each do |x|
+        arr << x if yield(x)
+      end
+    end
+    arr
+  end
+
+  def my_all?
+    return to_enum unless block_given?
+
+    my_each do |x|
+      return false unless yield(x)
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      my_each do |x|
+        return true if yield(x)
+      end
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum unless block_given?
+
+    my_each do |x|
+      return false if yield(x)
+    end
+    true
+  end
+
+  def my_count(element = nil)
+    total = 0
+    if block_given?
+      my_each do |x|
+        total += 1 if yield(x)
+      end
+    elsif !element.nil?
```

Comment:
This code looks similar to `elsif element` double negations is better avoided. `elsif element` returns true just as your code above does.